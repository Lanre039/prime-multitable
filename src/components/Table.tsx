import React from 'react';
import { ITableProps } from './types';
import './styles.css';

function Table(props: ITableProps): JSX.Element {
  const { primeNo } = props;

  //data = primeNo stored in the state (e.g 2,3, 5)
  const renderTd = (data: number[], rowNumber: number) =>
    data.map((cur, index) => <td key={index}>{rowNumber * cur}</td>);

  const renderTableData = () =>
    primeNo.map((rowNo, index) => {
      return (
        <tr key={index}>
          <th key={index}>{rowNo}</th>
          {renderTd(primeNo, rowNo)}
        </tr>
      );
    });

  return (
    <div
      className="table"
      style={{ visibility: primeNo.length ? 'visible' : 'hidden' }}
    >
      <table>
        <thead>
          <tr>
            <th></th>
            {primeNo.length > 0 ? (
              primeNo.map((data, index) => {
                return <th key={index}>{data}</th>;
              })
            ) : (
              <th>
                Enter positive integer greater than 1 only (e.g 2,3,4 etc.)
              </th>
            )}
          </tr>
        </thead>
        <tbody>{primeNo.length > 0 ? renderTableData() : null}</tbody>
      </table>
    </div>
  );
}

export default Table;
