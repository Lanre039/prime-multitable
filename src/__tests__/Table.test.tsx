import { render } from '@testing-library/react';
import Table from '../components/Table';

it('should take a snapshot', () => {
  const props = {
    primeNo: [],
  };
  const { asFragment } = render(<Table {...props} />);
  expect(asFragment()).toMatchSnapshot();
});
