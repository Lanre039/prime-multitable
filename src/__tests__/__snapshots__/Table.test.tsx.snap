// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="table"
    style="visibility: hidden;"
  >
    <table>
      <thead>
        <tr>
          <th />
          <th>
            Enter positive integer greater than 1 only (e.g 2,3,4 etc.)
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</DocumentFragment>
`;
